SPECS/maven/maven.spec

```diff
@@ -1,65 +1,109 @@
+# Switch "sources_generation" to 1 when running a package build to generate cached sources for regular builds.
+%define sources_generation 0
+%define m2_cache_tarball_name apache-%{name}-%{version}-m2.tar.gz
+%define licenses_tarball_name apache-%{name}-%{version}-licenses.tar.gz
+
+%if ! 0%{?sources_generation}
+%define offline_build -o
+%endif
+
 Summary:        Apache Maven
 Name:           maven
-Version:        3.5.4
-Release:        13%{?dist}
+Version:        3.8.1
+Release:        1%{?dist}
 License:        ASL 2.0
-URL:            https://maven.apache.org/
-Group:          Applications/System
 Vendor:         Microsoft Corporation
 Distribution:   Mariner
+Group:          Applications/System
+URL:            https://maven.apache.org/
 Source0:        https://archive.apache.org/dist/maven/maven-3/%{version}/source/apache-%{name}-%{version}-src.tar.gz
+# Using pre-compiled binaries because 'maven' requires itself during build-time.
 Source1:        https://archive.apache.org/dist/maven/maven-3/%{version}/binaries/apache-%{name}-%{version}-bin.tar.gz
-Source2:        apache-%{name}-%{version}-m2.tar.gz
-Source3:        apache-%{name}-%{version}-licenses.tar.gz
-# Note: this license tarball will need to be regenerated if this package is upgraded by running: apache-maven/src/main/appended-resources/META-INF/LICENSE.vm
-
+# In order to re-generate these sources after a version update, switch "sources_generation" to 1
+# and make sure network is enabled during the build. The tarballs will be inside the built 'maven-cached-sources' subpackage.
+%if ! 0%{?sources_generation}
+Source2:        %{m2_cache_tarball_name}
+Source3:        %{licenses_tarball_name}
+%endif
 BuildRequires:  ant
-BuildRequires:  openjre8
 BuildRequires:  openjdk8
+BuildRequires:  openjre8
 BuildRequires:  wget >= 1.15
```

Comment:
not in scope..but doe we even need that 1.15 qualifier?

Reply:
The build will be fine without it, but having additional information about the lower bound on build-time dependencies is a nice thing to have. Whether this particular line is currently true, however, is another thing.